src/pybloomfilter.pyx

```diff
@@ -43,37 +55,44 @@ cdef class BloomFilter:
     cdef cbloomfilter.BloomFilter * _bf
     cdef int _closed
     cdef int _in_memory
+    cdef int _oflags
     cdef public ReadFile
 
-    def __cinit__(self, capacity, error_rate, filename=None, perm=0755):
+    def __cinit__(self, capacity, error_rate, filename=None, mode="rw+", perm=0755):
         cdef char * seeds
         cdef long long num_bits
+        cdef int _capacity
+
         self._closed = 0
         self._in_memory = 0
+        self._oflags = os.O_RDWR
```

Comment:
Can we either rename it to `self._O_FLAGS` or add a note about what `_oflags` is? It took me a few minutes to figure it out!

Reply:
I am slightly wary of naming non-consts with uppercase.
I'll add an explanation above the definition?

Basically this needs to be set initially to `os.O_RDWR` before the `if filename is NoConstruct:` block is called, since otherwise, when copying a read only template, it would preserve the permissions ("r") and would not allow to create / write to the copy. Then, after `mode` is checked, I do `self._oflags = _construct_mode(mode)`.

Let me know if this approach seems okay to you.

Reply:
Well it was a nitpick since I didn't find anything else wrong with the changes here!

Don't bother with changing it just for the sake of it. Only need to figure out the merge conflict and let's get this and #14 out in v0.5.0.